dynawo/sources/Solvers/SolverIDA/DYNSolverIDA.cpp

```diff
@@ -676,15 +676,17 @@ SolverIDA::solveStep(double tAim, double &tNxt) {
   }
   std::sort(yErr.begin(), yErr.end(), mapcompabs());
 
-  Trace::debug() << DYNLog(SolverIDALargestErrors, nbErr) << Trace::endline;
-  vector<std::pair<double, int> >::iterator it;
-  int i = 0;
-  for (it = yErr.begin(); it != yErr.end(); ++it) {
+  if (yErr.size() > 0) {
+    Trace::debug() << DYNLog(SolverIDALargestErrors, nbErr) << Trace::endline;
+    vector<std::pair<double, int> >::iterator it;
+    int i = 0;
+    for (it = yErr.begin(); it != yErr.end(); ++it) {
```

Comment:
While we are modifying this I propose:
for (vector<std::pair<double, int> >::iterator it = yErr.begin(); it != yErr.end() && i<nbErr; ++it, ++i)
(and remove lines 685->687).

Reply:
done